tests/run_tests.py

```diff
@@ -128,10 +128,8 @@ def __init__(self, name, debug=False, optimize=False, mpi=False, openmp=False,
         if self.mpi:
             if os.path.exists(os.path.join(MPI_DIR, 'bin', 'mpifort')):
                 self.fc = os.path.join(MPI_DIR, 'bin', 'mpifort')
-            elif os.path.exists(os.path.join(MPI_DIR, 'bin', 'mpif90')):
-                self.fc = os.path.join(MPI_DIR, 'bin', 'mpif90')
             else:
-                raise RuntimeError('Cannot find an MPI Fortran compiler')
+                self.fc = os.path.join(MPI_DIR, 'bin', 'mpif90')
```

Comment:
Because all test configurations are created by default, and for some `Test()` is instantiated with `mpi=True`. Note that this is true even if you use the command-line `-C` flag to filter out MPI configurations.